src/controllers/MixcloudController.js

```diff
@@ -1,16 +1,37 @@
-controller = new BasicController({
-  supports: {
-    playpause: true,
-    previous: true,
-    next: true
+var isBeta = !!document.querySelector('.mz-player-control');
```

Comment:
I may be misunderstanding something about how the extension works. Does it not only execute the controller code once, every time the page is loaded? If it does, then as far as I can tell this should always work. The page never switches between using the old and new players without reloading the page, and the element that is matched by the selector is served up in the initial payload in both cases (as in it is not rendered to the DOM through javascript from a template, it's already there in the html that is sent to the client).

Looking through the `BasicController` code, I can see that `test` is only ever called during class construction, and it's done synchronously. Unless that constructor is called multiple times during the lifecycle of the page, it doesn't even matter if the DOM changes so that another `test` would have returned `true` because the `test`s are never run more than once anyway! This is why I don't get the usefulness of the `test` mechanism 🙂 .